packages/angular-table/src/index.ts

```diff
@@ -1,96 +1,61 @@
+export { FlexRenderDirective } from './flex-render'
 
 export function createAngularTable<TData extends RowData>(
-  options: () => TableOptions<TData>
-): Table<TData> & Signal<Table<TData>> {
-  const injector = inject(Injector)
-
-  return lazyInit(() =>
-    runInInjectionContext(injector, () => {
-      const resolvedOptionsSignal = computed<TableOptionsResolved<TData>>(
-        () => {
-          return {
-            state: {},
-            onStateChange: () => {},
-            renderFallbackValue: null,
-            ...options(),
-          }
-        }
-      )
-
-      const notifier = signal([], { equal: () => false })
-      const table = createTable(untracked(resolvedOptionsSignal))
-      const state = signal(table.initialState)
-
-      function updateOptions() {
-        const tableState = untracked(state)
-        const resolvedOptions = untracked(resolvedOptionsSignal)
-        untracked(() => {
-          table.setOptions(prev => ({
-            ...prev,
-            ...resolvedOptions,
-            state: { ...tableState, ...resolvedOptions.state },
-            onStateChange: updater => {
-              const value =
-                updater instanceof Function ? updater(tableState) : updater
-              state.set(value)
-              resolvedOptions.onStateChange?.(updater)
-            },
-          }))
-        })
-      }
-
-      updateOptions()
+  options: () => TableOptions<TData>,
+  injector?: Injector
+): Table<TData> {
+  if (!injector) injector = inject(Injector)
+
+  return lazyInit(() => {
+    // Compose table resolved options as computed.
+    // This will allow the effect to be triggered when options are updated.
+    const resolvedOptionsSignal = computed<TableOptionsResolved<TData>>(() => ({
+      state: {},
+      onStateChange: () => {},
+      renderFallbackValue: null,
+      ...options(),
+    }))
+
+    const table = createTable(resolvedOptionsSignal())
+
+    // By default, manage table state here using the table's initial state
+    const state = signal<TableState>(table.initialState)
+
+    function updateOptions() {
+      const tableState = state()
+      const resolvedOptions = resolvedOptionsSignal()
+      table.setOptions(prev => ({
+        ...prev,
+        ...resolvedOptions,
+        state: { ...tableState, ...resolvedOptions.state },
+        onStateChange: updater => {
+          const value =
+            updater instanceof Function ? updater(tableState) : updater
+          state.set(value)
+          resolvedOptions.onStateChange?.(updater)
+        },
+      }))
+    }
 
-      let firstRender = true
-      effect(() => {
-        void [state(), resolvedOptionsSignal()]
-        if (firstRender) {
-          return (firstRender = false)
-        }
-        untracked(() => {
-          updateOptions()
-          notifier.set([])
-        })
-      })
+    // set table options again when options are updated
+    effect(() => updateOptions(), { injector })
```

Comment:
Consider that now this portion of code was present in order to avoid the twich `setOptions` call during the first table initialization.

<img width="581" alt="Screenshot 2024-05-05 alle 19 54 09" src="https://github.com/TanStack/table/assets/37072694/c3135749-29e1-41aa-bc03-0e75c870c61d">

Reply:
'updateOptions' will be triggered twice everytime options is changed from outside. This is because 'effect' is listening to 2 signals. I was thinking of handling in all scenarios, if we want to fix this, instead of handling only the initialisation part.

Reply:
Yeah you're right. I've also seen that currently the svelte adapter has the issue, since it's subscribing to each options and states.